provide-reactor.js

```diff
@@ -0,0 +1,74 @@
+var React = require('react')
+var hoistNonReactStatics = require('hoist-non-react-statics')
+var objectAssign = require('object-assign')
+
+function createComponent(Component, additionalContextTypes) {
+  var childContextTypes = objectAssign({
+    reactor: React.PropTypes.object.isRequired,
+  }, additionalContextTypes || {})
+
+  var ReactorProvider = React.createClass({
+    displayName: 'ReactorProvider',
+
+    propTypes: {
+      reactor: React.PropTypes.object.isRequired,
+    },
+
+    childContextTypes: childContextTypes,
+
+    getChildContext: function() {
+      var childContext = {
+        reactor: this.props.reactor,
+      }
+      if (additionalContextTypes) {
+        Object.keys(additionalContextTypes).forEach(function(key) {
+          childContext[key] = this.props[key]
+        }, this)
+      }
+      return childContext
+    },
+
+    render: function() {
+      return React.createElement(Component, this.props)
+    },
+  })
+
+  hoistNonReactStatics(ReactorProvider, Component)
+
+  return ReactorProvider
+}
+
+/**
+ * Provides reactor prop to all children as React context
+ *
+ * Example:
+ *   var WrappedComponent = provideReactor(Component, {
+ *       foo: React.PropTypes.string
+ *   });
+ *
+ * Also supports the decorator pattern:
+ *   @provideReactor({
+ *       foo: React.PropTypes.string
+ *   })
+ *   class BaseComponent extends React.Component {
+ *       render() {
+ *           return <div/>;
+ *       }
+ *   }
+ *
+ * @method provideReactor
+ * @param {React.Component} [Component] component to wrap
+ * @param {object} additionalContextTypes Additional contextTypes to add
+ * @returns {React.Component} or {Function} if using decorator pattern
+ */
+module.exports = function provideReactor(Component, additionalContextTypes) {
+  // support decorator pattern
+  if (arguments.length === 0 || typeof arguments[0] !== 'function') {
```

Comment:
(comment on outdated diff).

And it's the pattern, if it's a function we are simply wrapping the React.Element directly